O/openmp/build_tarballs.jl

```diff
@@ -0,0 +1,43 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder
+
+name = "openmp"
+version = v"12.0.0"
+
+# Collection of sources required to build mpg123
+sources = [
+    ArchiveSource(
+        "https://github.com/llvm/llvm-project/releases/download/llvmorg-$version/openmp-$version.src.tar.xz",
+        "eb1b7022a247332114985ed155a8fb632c28ce7c35a476e2c0caf865150f167d"
+    ),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/openmp-*/
+
+if [[ "${target}" == *-freebsd* ]]; then
+    CMAKE_SHARED_LINKER_FLAGS="-Wl,--version-script=$(pwd)/runtime/src/exports_so.txt"
+fi
+cmake . -DCMAKE_INSTALL_PREFIX=${prefix} -DCMAKE_TOOLCHAIN_FILE="${CMAKE_TARGET_TOOLCHAIN}" -DCMAKE_SHARED_LINKER_FLAGS="${CMAKE_SHARED_LINKER_FLAGS}" -DLIBOMP_INSTALL_ALIASES=OFF
+make -j${nproc}
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+# disable for mingw for now, blocking on uasm
+platforms = expand_cxxstring_abis(supported_platforms(exclude=Sys.iswindows))
```

Comment:
What's the problem with Windows?

Reply:
As mentioned above, there's no suitable ASM compiler (mingw uses uasm)

> I struggled to make it work on freebsd, but it's hard to compile with mingw, you could see in official [code](https://github.com/msys2/MINGW-packages/blob/d440dcb738718a7b73ea1de191a03a0f9c201244/mingw-w64-clang/PKGBUILD#L297-L298), it relies on [uasm](https://github.com/Terraspace/UASM), which I'd like to have it in BinaryBuilderBase, I fired a issue [here](https://github.com/JuliaPackaging/BinaryBuilderBase.jl/issues/161) (but nobody replies).

Reply:
https://github.com/JuliaPackaging/BinaryBuilderBase.jl/issues/161#issuecomment-877643370